src/setup/node.rs

```diff
@@ -43,37 +45,90 @@ pub enum ChildExitCode {
     ErrorCode(Option<i32>),
 }
 
+/// Node type is used to select different startup configurations.
+pub enum NodeType {
+    /// A temporary node used to store ledger data for stateful nodes. Should not be used otherwise.
+    Testnet,
+    /// A non-validator node without any preloaded data.
+    Stateless,
+    /// A validator node with a preloaded ledger data.
+    Stateful,
+}
+
 pub struct NodeBuilder {
     conf: NodeConfig,
     meta: NodeMetaData,
 }
 
 impl NodeBuilder {
-    /// Creates new [NodeBuilder]. Initial state is taken from `state` path, the node will run in `target` path.
-    pub fn new(state: Option<PathBuf>, target: PathBuf) -> Result<Self> {
-        if !target.exists() {
-            fs::create_dir_all(&target)?;
-        }
-        let mut copy_options = CopyOptions::new();
-        copy_options.content_only = true;
-        copy_options.overwrite = true;
+    /// Creates new [NodeBuilder].
+    pub fn new() -> anyhow::Result<Self> {
         let setup_path = build_ripple_work_path()?.join(RIPPLE_SETUP_DIR);
-        copy(&setup_path, &target, &copy_options)?;
-        if let Some(source) = state {
-            copy(&source, &target, &copy_options)?;
-        }
-        let conf = NodeConfig::new(target);
-        let meta = NodeMetaData::new(conf.path.clone())?;
+
+        let conf = NodeConfig::default();
+        let meta = NodeMetaData::new(setup_path)?;
         Ok(Self { conf, meta })
     }
 
+    /// Creates default [NodeBuilder] which can handle stateful nodes.
+    pub fn default() -> anyhow::Result<Self> {
```

Comment:
actually, this should be `stateful` (and the other should probably be `stateless`; also, `default` implies that `Self` is returned (not a `Result`)